### PR TITLE
Education Benefits Wizard

### PR DESCRIPTION
**Pull request by maharielrosario**
## Description
This PR  uses the 1990 form as a basis for creating a reusable wizard for education-benefit forms and adds some functionality to the static pages wizard to support setting the wizard status and the benefit referred.

## Testing done
- Unit tests have been created for the static pages wizard module as there were none created yet
- Unit tests have been created for the WizardContainer since it's a new component
- Unit tests for the 1990 Introduction Page have been updated
- E2E tests have been updated

## Screenshots
screenshot-localhost_3001-2020 07 31-10_22_31
screenshot-localhost_3001-2020 07 31-10_22_31 (1)

Screen Shot 2020-07-31 at 10 24 52 AM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/edu-benefits/tests/wizard/containers/WizardContainer.unit.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import WizardContainer from '../../../wizard/containers/WizardContainer';
```
Comment (bradl3yC) [inline src/applications/edu-benefits/tests/wizard/containers/WizardContainer.unit.spec.jsx:2]: need to swap all of these to relative imports

File: src/applications/static-pages/tests/wizard/edu-benefits/index.unit.test.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  it('should initially render the wizard start button', () => {
+    const wrapper = shallow(<Wizard {...defaultProps} />);
+    expect(wrapper.exists('.wizard-button')).to.equal(true);
+    wrapper.unmount();
+  });
+  it('should take you to the 1990E no warning site', () => {
+    const wrapper = mount(<Wizard {...defaultProps} />);
+    const instance = wrapper.instance();
+    wrapper.find('.wizard-button').simulate('click');
+    // eslint-disable-next-line no-unused-expressions
```
[the remark this answers is not in the document]
Comment (mr0sari0) [reply src/applications/static-pages/tests/wizard/edu-benefits/index.unit.test.jsx:64]: I get the following eslint error: `Expected an assignment or function call and instead saw an expression. eslint (no-unused-expressions)` when I use `expect(wrapper.state('pageHistory')[0].state).to.be.undefined;`

File: src/applications/static-pages/tests/wizard/edu-benefits/index.unit.test.jsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    expect(wrapper.state('pageHistory')[2].state).to.deep.equal({
+      selected: 'no',
+    });
+    wrapper.find('#TransferredBenefits-1').invoke('onChange')({
+      target: { value: 'yes' },
+    });
+    expect(wrapper.state('pageHistory')[3].state).to.deep.equal({
+      selected: 'yes',
+    });
+    // eslint-disable-next-line no-unused-expressions
```
[the remark this answers is not in the document]
Comment (bradl3yC) [reply src/applications/static-pages/tests/wizard/edu-benefits/index.unit.test.jsx:90]: @mr0sari0 can you link where you read this is a bug?

[the remark this answers is not in the document]
Comment (mr0sari0) [reply src/applications/static-pages/tests/wizard/edu-benefits/index.unit.test.jsx]: @bradl3yC Here you go https://github.com/cypress-io/eslint-plugin-cypress/issues/3#issuecomment-363160343

I just talked with VSP and it was actually because the file needed to end with `unit.spec.jsx` rather than `unit.test.jsx`.  Apparently that is set up in the .eslintrc file. 

 https://dsva.slack.com/archives/CBU0KDSB1/p1596207575386300?thread_ts=1596207246.384600&cid=CBU0KDSB1

I just did a refactor and got rid of the eslint disabling.

**Review by cohnjesse — APPROVED** [review]
LGTM

File: src/applications/edu-benefits/components/GetFormHelp.jsx [hunk 4]
```diff
@@ -5,27 +5,16 @@ function GetFormHelp() {
...
-        <br />
-        ask the Education Call Center:
-      </p>
-      <p className="help-phone-number">
-        <a className="help-phone-number-link" href="tel:1-888-442-4551">
-          888-442-4551 (888-GI-BILL-1)
+        Call us at{' '}
+        <a href="tel:800-827-1000" aria-label="8 0 0. 8 2 7. 1 0 0 0.">
+          800-827-1000
         </a>
```
Comment (Mottie) [inline src/applications/edu-benefits/components/GetFormHelp.jsx:11]: The `<Telephone />` component could be used here.

File: src/applications/edu-benefits/components/GetFormHelp.jsx [hunk 5]
```diff
@@ -5,27 +5,16 @@ function GetFormHelp() {
...
-        <br />
-        Monday &#8211; Friday, 8:00 a.m. &#8211; 7:00 p.m. ET
-        <br />
-        <a
-          className="help-phone-number-link"
-          href="https://gibill.custhelp.com/app/utils/login_form/redirect/ask"
-        >
-          Submit a question to Education Service
+        . We’re here Monday through Friday, 8:00 am to 9:00 pm ET. If you have
```
Comment (Mottie) [inline src/applications/edu-benefits/components/GetFormHelp.jsx:12]: Please use "a.m." and "p.m." here ([ref](https://design.va.gov/content-style-guide/dates-and-numbers#times-and-time-zones)). Also, screenreaders would say "am" and "pm" as words and not separate letters.

File: src/applications/edu-benefits/components/GetFormHelp.jsx [hunk 6]
```diff
@@ -5,27 +5,16 @@ function GetFormHelp() {
...
+        hearing loss, call{' '}
+        <a href="tel:711" aria-label="TTY. 7 1 1.">
+          TTY: 711
         </a>
```
Comment (Mottie) [inline src/applications/edu-benefits/components/GetFormHelp.jsx:16]: `<Telephone contact={CONTACTS['711']} pattern={PATTERNS['911']} />`

File: src/applications/edu-benefits/components/GetFormHelp.jsx [hunk 7]
```diff
@@ -5,27 +5,16 @@ function GetFormHelp() {
...
-        For help filling out this form,
-        <br />
-        ask the Education Call Center:
-      </p>
-      <p className="help-phone-number">
-        <a className="help-phone-number-link" href="tel:1-888-442-4551">
-          888-442-4551 (888-GI-BILL-1)
+        Call us at{' '}
+        <a href="tel:800-827-1000" aria-label="8 0 0. 8 2 7. 1 0 0 0.">
+          800-827-1000
```
Comment (Mottie) [inline src/applications/edu-benefits/components/GetFormHelp.jsx:10]: Why the change from `888-442-4551` to `800-827-1000`? Please make sure this was an intended change.

File: src/applications/edu-benefits/components/GetFormHelp.jsx [hunk 8]
```diff
@@ -5,27 +5,16 @@ function GetFormHelp() {
...
+        . We’re here Monday through Friday, 8:00 am to 9:00 pm ET. If you have
+        hearing loss, call{' '}
+        <a href="tel:711" aria-label="TTY. 7 1 1.">
+          TTY: 711
         </a>
       </p>
-      <p className="help-talk">
-        If this form isn't working right for you, please <CallHRC />
-      </p>
     </div>
```
Comment (Mottie) [inline src/applications/edu-benefits/components/GetFormHelp.jsx:18]: Was this supposed to be removed?

File: src/applications/edu-benefits/sass/edu-benefits.scss [hunk 9]
```diff
@@ -140,6 +140,24 @@
...
+  margin-left: 0.5em;
+  margin-right: -0.3em;
+  text-indent: -9999em;
+  width: 1em;
+}
+
+.wizard-button[aria-expanded=true]:after {
+  background-image: url(/img/arrow-up-white.svg);
+}
+
```
Comment (Mottie) [inline src/applications/edu-benefits/sass/edu-benefits.scss:160]: These css additions should probably be added to the platform styles since it'll be used by all wizards eventually.

File: src/applications/edu-benefits/tests/wizard/containers/WizardContainer.unit.spec.jsx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    sessionStorage: {},
+  };
+  let setWizardStatus;
+
+  before(() => {
+    global.sessionStorage = {
+      getItem: key =>
+        key in mockStore.sessionStorage ? mockStore.sessionStorage[key] : null,
+      setItem: (key, value) => {
+        mockStore.sessionStorage[key] = `${value}`;
```
Comment (Mottie) [inline src/applications/edu-benefits/tests/wizard/containers/WizardContainer.unit.spec.jsx:23]: `mockStore.sessionStorage[key] = value;`

Also, it might be easier to add this duplicate `sessionStorage` code into a `util` file and import it.

File: src/applications/edu-benefits/tests/1990/containers/IntroductionPage.unit.spec.jsx [hunk 11]
```diff
@@ -1,30 +1,69 @@
...
+  };
+
+  let defaultProps;
+
+  before(() => {
+    global.sessionStorage = {
+      getItem: key =>
+        key in mockStore.sessionStorage ? mockStore.sessionStorage[key] : null,
+      setItem: (key, value) => {
+        mockStore.sessionStorage[key] = `${value}`;
```
Comment (Mottie) [inline src/applications/edu-benefits/tests/1990/containers/IntroductionPage.unit.spec.jsx:23]: `mockStore.sessionStorage[key] = value;`

File: src/applications/edu-benefits/tests/wizard/containers/WizardContainer.unit.spec.jsx (hunk 1 above)
Comment (mr0sari0) [reply src/applications/edu-benefits/tests/wizard/containers/WizardContainer.unit.spec.jsx]: @bradl3yC Just updated.

File: src/applications/edu-benefits/wizard/containers/WizardContainer.jsx [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+        href="#"
+        className="vads-u-display--inline-block vads-u-margin-bottom--3 skip-wizard-link"
+        onClick={e => {
+          e.preventDefault();
+          setReferredBenefit(currentFormIdSuffix);
+          setWizardStatus(WIZARD_STATUS_COMPLETE);
+        }}
+      >
+        Apply online with VA Form {currentVAFormId}
+      </a>
```
Comment (Mottie) [inline src/applications/edu-benefits/wizard/containers/WizardContainer.jsx:70]: Instead of using a link (`<a>`) here, please change this to a button with a class of `va-button-link`

File: src/applications/static-pages/wizard/index.js [hunk 13]
```diff
@@ -3,16 +3,40 @@ import PropTypes from 'prop-types';
...
+export const formIdSuffixes = {
+  FORM_ID_1990: '1990',
+  FORM_ID_10203: '10203',
+  FORM_ID_1995: '1995',
+  FORM_ID_0994: '0994',
+  FORM_ID_5495: '5495',
+  FORM_ID_5490: '5490',
+  FORM_ID_1990E: '1990E',
+  FORM_ID_1990N: '1990N',
+};
```
Comment (Mottie) [inline src/applications/static-pages/wizard/index.js:21]: Should all these constants be added to this file? I'd think they are edu-benefits specific and should be included in that folder.

Comment (Mottie) [reply src/applications/static-pages/wizard/index.js:21]: Same idea for the `NO_BENEFIT_REFERRED` constant.

File: src/applications/static-pages/wizard/index.js [hunk 14]
```diff
@@ -3,16 +3,40 @@ import PropTypes from 'prop-types';
...
 export default class Wizard extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       pageHistory: [props.pages[0]],
       currentPageIndex: 0,
       expanded: !props.expander,
+      benefitReferred: getReferredBenefit(),
+      wizardStatus: getWizardStatus(),
     };
```
Comment (Mottie) [inline src/applications/static-pages/wizard/index.js:38]: `benefitReferred` should be more generic (have a more generic name) to suit all services that use the wizard.

**Review by Mottie — CHANGES_REQUESTED** [review]
Wow, that's a lot of work. Good job!

**Review by Clark-W-Griswold — APPROVED** [review]
👍

File: src/applications/edu-benefits/components/GetFormHelp.jsx (hunk 4 above)
Comment (mr0sari0) [reply src/applications/edu-benefits/components/GetFormHelp.jsx]: Just refactored to include it!

File: src/applications/edu-benefits/components/GetFormHelp.jsx (hunk 5 above)
Comment (mr0sari0) [reply src/applications/edu-benefits/components/GetFormHelp.jsx]: Just refactored to include this.

File: src/applications/edu-benefits/components/GetFormHelp.jsx (hunk 6 above)
Comment (mr0sari0) [reply src/applications/edu-benefits/components/GetFormHelp.jsx]: Applied.

File: src/applications/edu-benefits/components/GetFormHelp.jsx (hunk 7 above)
Comment (mr0sari0) [reply src/applications/edu-benefits/components/GetFormHelp.jsx]: This was a change specified by the content team [in this comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/11550#issuecomment-664603173).

File: src/applications/edu-benefits/components/GetFormHelp.jsx (hunk 8 above)
Comment (mr0sari0) [reply src/applications/edu-benefits/components/GetFormHelp.jsx:17]: Yes. Please refer to the comment I linked to.

File: src/applications/edu-benefits/tests/wizard/containers/WizardContainer.unit.spec.jsx (hunk 10 above)
Comment (mr0sari0) [reply src/applications/edu-benefits/tests/wizard/containers/WizardContainer.unit.spec.jsx]: Just created a `util` file to hold that code.

File: src/applications/edu-benefits/wizard/containers/WizardContainer.jsx (hunk 12 above)
Comment (mr0sari0) [reply src/applications/edu-benefits/wizard/containers/WizardContainer.jsx]: Just updated.

File: src/applications/static-pages/wizard/index.js (hunk 13 above)
Comment (Mottie) [reply src/applications/static-pages/wizard/index.js:21]: Ok, I guess as long as @Clark-W-Griswold is cool with it 👍

File: src/applications/edu-benefits/components/GetFormHelp.jsx [hunk 15]
```diff
@@ -1,30 +1,18 @@
...
-          className="help-phone-number-link"
-          href="https://gibill.custhelp.com/app/utils/login_form/redirect/ask"
-        >
-          Submit a question to Education Service
-        </a>
-      </p>
-      <p className="help-talk">
-        If this form isn't working right for you, please <CallHRC />
+        Call us at <Telephone contact={CONTACTS.VA_BENEFITS} />. We’re here
+        Monday through Friday, 8:00 a.m to 9:00 p.m ET. If you have hearing
```
Comment (Mottie) [inline src/applications/edu-benefits/components/GetFormHelp.jsx:13]: "a.m." & "p.m." - the last period is missing on each.

File: src/applications/edu-benefits/components/GetFormHelp.jsx (hunk 7 above)
Comment (Mottie) [reply src/applications/edu-benefits/components/GetFormHelp.jsx]: Ok, as long as @DanielleThierryUSDSVA is aware.

**Review by Mottie — APPROVED** [review]

File: src/applications/edu-benefits/components/GetFormHelp.jsx (hunk 15 above)
Comment (mr0sari0) [reply src/applications/edu-benefits/components/GetFormHelp.jsx:13]: Just updated.

**Review by bradl3yC — APPROVED** [review]
